tests/integration/test_changes.py

```diff
@@ -0,0 +1,103 @@
+import pybop
+import pytest
+import numpy as np
+
+
+class TestChanges:
+    """
+    A class to test that different inputs return different outputs.
+    """
+
+    @pytest.fixture(
+        params=[
+            pybop.Parameter(  # geometric parameter
+                "Negative particle radius [m]",
+                prior=pybop.Gaussian(6e-06, 0.1e-6),
+                bounds=[1e-6, 9e-6],
+                true_value=5.86e-6,
+            ),
+            pybop.Parameter(  # non-geometric parameter
+                "Positive electrode diffusivity [m2.s-1]",
+                prior=pybop.Gaussian(3.43e-15, 1e-15),
+                bounds=[1e-15, 5e-15],
+                true_value=4e-15,
+            ),
+        ]
+    )
+    def parameter(self, request):
+        return request.param
+
+    @pytest.mark.integration
+    def test_changing_experiment(self, parameter):
+        # Change the experiment and check that the results are different.
+
+        parameter_set = pybop.ParameterSet.pybamm("Chen2020")
+        parameters = [parameter]
+        init_soc = 0.5
+        model = pybop.lithium_ion.SPM(parameter_set=parameter_set)
+
+        t_eval = [0, 3600]  # Default 1C discharge to cut-off voltage
```

Comment:
I believe this will result in the prediction output to be of length 100. I think this stems from the default result pybamm provides if a sampling time is not requested. To ensure the assert below is purely based on element differences, I think we want something like this instead:
```suggestion
        t_eval = np.arange(0, 3600, 2)  # Default 1C discharge to cut-off voltage
```

To ensure the shapes are always the same, it might be a good idea to shrink the simulation time to be a bit below 3600.